File: airbyte-connector-builder-server/src/main/java/io/airbyte/connectorbuilder/exceptions/ConnectorBuilderException.java
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.connectorbuilder.exceptions;

import io.airbyte.commons.server.errors.KnownException;
import io.micronaut.http.HttpStatus;

/**
 * Thrown when the Connector Builder encountered an error when processing the request.
 */
public class ConnectorBuilderException extends KnownException {

  public ConnectorBuilderException(final String message, final Exception exception) {
    super(message, exception);
  }

  public ConnectorBuilderException(final String message) {
    super(message);
  }

  @Override
  public int getHttpCode() {
    return HttpStatus.INTERNAL_SERVER_ERROR.getCode();
  }

}

File: airbyte-commons-server/src/main/kotlin/io/airbyte/commons/server/builder/exceptions/ConnectorBuilderException.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.server.builder.exceptions

import io.airbyte.commons.server.errors.KnownException
import io.micronaut.http.HttpStatus

/**
 * Thrown when the Connector Builder encountered an error when processing the request.
 */
class ConnectorBuilderException : KnownException {
  constructor(message: String, exception: Exception) : super(message, exception)

  constructor(message: String) : super(message)

  override fun getHttpCode(): Int = HttpStatus.INTERNAL_SERVER_ERROR.code
}
